controllers/harborconfiguration_controller.go

```diff
@@ -390,20 +390,23 @@ func getConcreteHarborType(ctx context.Context, crdClient dynamic.ResourceInterf
 }
 
 func deleteAll(ctx context.Context, harborConfiguration harborconfigurationv1alpha1.HarborConfiguration, client *apiv2.RESTClient) (ctrl.Result, error) {
-	err := client.DeleteRegistryByID(ctx, harborConfiguration.Status.RegistryId)
+	deletionLog := ctrl.Log.WithName("deletion")
+
+	err := client.DeleteReplicationPolicyByID(ctx, harborConfiguration.Status.ReplicationId)
 	if err != nil {
```

Comment:
Should we check if the error is something along the lines of `notFound`? It would then indicate that the we have already deleted it

Reply:
Yeah we can make this change if you would like.

Reply:
Actually doing a fairly big refactor here and using get methods to obtain the information for objects we are trying to delete. The reason why is because status wasn't updating with the ids we needed.